src/main/java/eu/dissco/dataexporter/service/DataExporterService.java

```diff
@@ -1,8 +1,69 @@
 package eu.dissco.dataexporter.service;
 
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import eu.dissco.dataexporter.database.jooq.enums.ExportType;
+import eu.dissco.dataexporter.database.jooq.enums.JobState;
+import eu.dissco.dataexporter.domain.ExportJob;
+import eu.dissco.dataexporter.exception.InvalidRequestException;
+import eu.dissco.dataexporter.repository.DataExporterRepository;
+import eu.dissco.dataexporter.schema.ExportJobRequest;
+import java.security.MessageDigest;
+import java.time.Instant;
+import java.util.UUID;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
 import org.springframework.stereotype.Service;
 
 @Service
+@Slf4j
+@RequiredArgsConstructor
 public class DataExporterService {
 
+  private final DataExporterRepository repository;
+  private final ObjectMapper mapper;
+  private final MessageDigest messageDigest;
+
+  public void addJobToQueue(ExportJobRequest jobRequest, String orcid)
+      throws InvalidRequestException {
+    var timestamp = Instant.now();
+    var params = mapper.valueToTree(jobRequest.getData().getAttributes().getParams().getAdditionalProperties());
```

Comment:
The flow for when the params are an exact match with existing export is later right?

Reply:
not in this pr, was going to make a separate user story for that but i know how it'll go

Reply:
Excellent